src/lm.jl

```diff
@@ -156,7 +156,32 @@ function coeftable(mm::LinearModel)
               ["x$i" for i = 1:size(mm.pp.X, 2)], 4)
 end
 
-predict(mm::LinearModel, newx::Matrix) =  newx * coef(mm)
+predict(mm::LinearModel, newx::AbstractMatrix) = newx * coef(mm)
+
+"""
+For linear models, specifying `interval_type` will return a 3-tuple with the
+predicted values, the lower and the higher confidence bound. Confidence intervals
+delimit the uncertainty of the estimate of the predicted values, prediction
+intervals delimit the estimated bounds for any new data points from the
+population.
+
+# Arguments
+* `interval_type::Symbol`: one of `:confint` or `:predint`
+* `level=0.95`: the level of the confidence interval
+* `alpha=1-level`: an alternative way of specifying the level of the interval
```

Comment:
I think providing two arguments for the same thing is actually more confusing than useful. Computing `level` from `alpha` is really simple anyway.

Then better document arguments directly in the description paragraph since they are only two. They could very naturally be inserted into existing sentences.